apps/ui/src/helpers/utils.ts

```diff
@@ -610,8 +610,11 @@ export function getSocialNetworksLink(data: any) {
 export function getFormattedVotingPower(votingPower?: VotingPowerItem) {
   if (!votingPower) return;
 
-  const { totalVotingPower, decimals, symbol } = votingPower;
-  const value = _vp(Number(totalVotingPower) / 10 ** decimals);
+  const { votingPowers, symbol } = votingPower;
+
+  const value = _vp(
+    votingPowers.reduce((acc, b) => acc + Number(b.value) / 10 ** b.decimals, 0)
+  );
```

Comment:
It's going to work with onchain spaces.

On chain decimals are only used for displaying values. Cumulative VP is just sum of raw values.

If I have OZ Votes strategy (that uses 18 decimals) and I have 1 such token then my displayed VP will be 1, but underlying value is `1000000000000000000`.

If I now also add whitelist and whitelist gives me 1 raw voting power my new VP is `1000000000000000000`, but with this change it would have been displayed as 2 VP on UI.

Reply:
Before:
<img width="477" alt="image" src="https://github.com/user-attachments/assets/bc4ea31f-339f-4161-988c-b48c39ab9499">

After:
<img width="471" alt="image" src="https://github.com/user-attachments/assets/24144ffa-e21d-46df-973d-3138b3122287">

Bold value is "raw" value, non bold is per-strategy formatted value.

Reply:
Could be approached like this (not sure I like naming of `decimals`/`finalDecimals`), we can come up with something better):
```diff
diff --git a/apps/ui/src/components/Modal/VotingPower.vue b/apps/ui/src/components/Modal/VotingPower.vue
index 4ef08e76..1bc81256 100644
--- a/apps/ui/src/components/Modal/VotingPower.vue
+++ b/apps/ui/src/components/Modal/VotingPower.vue
@@ -58,10 +58,14 @@ const loading = computed(
           />
           <div class="text-skin-link shrink-0">
             {{
-              _n(Number(strategy.value) / 10 ** strategy.decimals, 'compact', {
-                maximumFractionDigits: 2,
-                formatDust: true
-              })
+              _n(
+                Number(strategy.value) / 10 ** strategy.finalDecimals,
+                'compact',
+                {
+                  maximumFractionDigits: 2,
+                  formatDust: true
+                }
+              )
             }}
             {{ votingPower.symbol }}
           </div>
diff --git a/apps/ui/src/helpers/utils.ts b/apps/ui/src/helpers/utils.ts
index d4645094..68e00c3a 100644
--- a/apps/ui/src/helpers/utils.ts
+++ b/apps/ui/src/helpers/utils.ts
@@ -613,7 +613,10 @@ export function getFormattedVotingPower(votingPower?: VotingPowerItem) {
   const { votingPowers, symbol } = votingPower;
 
   const value = _vp(
-    votingPowers.reduce((acc, b) => acc + Number(b.value) / 10 ** b.decimals, 0)
+    votingPowers.reduce(
+      (acc, b) => acc + Number(b.value) / 10 ** b.finalDecimals,
+      0
+    )
   );
 
   return symbol ? `${value} ${symbol}` : value;
diff --git a/apps/ui/src/networks/evm/actions.ts b/apps/ui/src/networks/evm/actions.ts
index 1445961e..69bcba80 100644
--- a/apps/ui/src/networks/evm/actions.ts
+++ b/apps/ui/src/networks/evm/actions.ts
@@ -736,11 +736,22 @@ export function createActions(
       if (snapshotInfo.at === null)
         throw new Error('EVM requires block number to be defined');
 
+      const finalDecimals = Math.max(
+        ...strategiesMetadata.map(metadata => metadata.decimals ?? 0)
+      );
+
       return Promise.all(
         strategiesAddresses.map(async (address, i) => {
           const strategy = getEvmStrategy(address, networkConfig);
           if (!strategy)
-            return { address, value: 0n, decimals: 0, token: null, symbol: '' };
+            return {
+              address,
+              value: 0n,
+              decimals: 0,
+              finalDecimals,
+              token: null,
+              symbol: ''
+            };
 
           const strategyMetadata = await parseStrategyMetadata(
             strategiesMetadata[i].payload
@@ -762,6 +773,7 @@ export function createActions(
             address,
             value,
             decimals: strategiesMetadata[i]?.decimals ?? 0,
+            finalDecimals,
             symbol: strategiesMetadata[i]?.symbol ?? '',
             token,
             swapLink: getSwapLink(strategy.type, address, chainId)
diff --git a/apps/ui/src/networks/types.ts b/apps/ui/src/networks/types.ts
index 6fcfcaf7..f32a79b3 100644
--- a/apps/ui/src/networks/types.ts
+++ b/apps/ui/src/networks/types.ts
@@ -111,6 +111,7 @@ export type VotingPower = {
   address: string;
   value: bigint;
   decimals: number;
+  finalDecimals: number;
   token: string | null;
   symbol: string;
   chainId?: number;
```

Reply:
Explanation: we add separation between per-strategy decimals and final decimals that need to be used to sum up values. In case of onchain spaces we compute `finalDecimals` as max decimals from all strategies. In case of offchain spaces we just use strategy decimals as there is no this distinction.

This diff doesn't include all changes (we should have similar approach for Starknet, offchain needs to define finalDecimals as same thing as decimals, SpaceUser should use finalDecimals as well I think).